python/triton/runtime/cache.py

```diff
@@ -41,7 +41,7 @@ def __init__(self, key):
         self.lock_path = None
         # create cache directory if it doesn't exist
         self.cache_dir = os.environ.get('TRITON_CACHE_DIR', default_cache_dir())
-        if self.cache_dir:
+        if not self.cache_dir:
```

Comment:
hmm, I think the "fix" is somehow not as expected?

`os.environ.get` get a default dir anyway, and we use `exist_ok` to skip altering the existing dir.
So we can just remove the `if self.cache_dir:` condition

Reply:
Oh right makes sense. I think the bug was something else and not executing this branch actually fixed it. Let me take a look tmr.

Reply:
The bug I faced was that `put` would return None

Reply:
Turns out the issue was not handling empty env var (i.e. ""). PTAL.